### ISSUE TITLE
Update to SF (v2) but commands are working with SFDX and not working with SF 

### ISSUE THREAD

**Issue by fedegiust** [forcedotcom/cli#2389]
### Summary
This is an issue on my local env.
I'm using Windows 10 with git bash and sfdx cli v2
Basically when trying to run the new commands with ``sf`` I get an error but they work with ``sfdx``

![image](https://github.com/forcedotcom/cli/assets/1804862/9308f472-30e9-41c7-96f8-b32164ea6f01)

![image](https://github.com/forcedotcom/cli/assets/1804862/81396fdd-118f-4550-9154-c0467cea5edc)

![image](https://github.com/forcedotcom/cli/assets/1804862/8f22e5fa-c9bc-4293-b4b1-83152ed30ab4)

![image](https://github.com/forcedotcom/cli/assets/1804862/baa42e35-222e-4e9e-978e-4c4982c87173)

### Steps To Reproduce

Not sure how to reproduce, what I did for the upgrade was unintalling the SFDX version that I had on this laptop and installed SF CLI using the latest .exe

### Expected result
I guess commands should work with both ``sf`` and ``sfdx`` as far as I understood

### Actual result
Getting errors ``is not a sf command`` when trying to use ``sf``

### System Information
I'm using git bash

```json
{
  "cliVersion": "@salesforce/cli/2.1.7",
  "architecture": "win32-x64",
  "nodeVersion": "node-v18.15.0",
  "osVersion": "Windows_NT 10.0.19045",
  "shell": "C:\\Program Files\\Git\\usr\\bin\\bash.exe",
  "rootPath": "C:\\Users\\Usuario\\AppData\\Local\\sf\\client\\2.1.7-355d833",
  "pluginVersions": [
    "@oclif/plugin-autocomplete 2.3.2 (core)",
    "@oclif/plugin-commands 2.2.18 (core)",
    "@oclif/plugin-help 5.2.13 (core)",
    "@oclif/plugin-not-found 2.3.31 (core)",
    "@oclif/plugin-plugins 3.1.6 (core)",
    "@oclif/plugin-search 0.0.20 (core)",
    "@oclif/plugin-update 3.1.26 (core)",
    "@oclif/plugin-version 1.3.7 (core)",
    "@oclif/plugin-warn-if-update-available 2.0.43 (core)",
    "@oclif/plugin-which 2.2.25 (core)",
    "@salesforce/cli 2.1.7 (core)",
    "apex 2.3.6 (core)",
    "auth 2.8.9 (core)",
    "data 2.5.1 (core)",
    "deploy-retrieve 1.16.0 (core)",
    "info 2.6.31 (core)",
    "limits 2.3.26 (core)",
    "login 1.2.20 (core)",
    "org 2.9.25 (core)",
    "schema 2.3.20 (core)",
    "settings 1.4.20 (core)",
    "sobject 0.1.37 (core)",
    "source 2.10.25 (core)",
    "telemetry 2.2.3 (core)",
    "templates 55.5.5 (core)",
    "trust 2.4.32 (core)",
    "user 2.3.24 (core)",
    "sfdmu 4.29.5 (user)",
    "sfdx-git-delta 5.24.1 (user)"
  ]
}
```
### Additional information
sfdx doctor output
```json
{
  "versionDetail": {
    "cliVersion": "@salesforce/cli/2.1.7",
    "architecture": "win32-x64",
    "nodeVersion": "node-v18.15.0",
    "osVersion": "Windows_NT 10.0.19045",
    "shell": "C:\\Program Files\\Git\\usr\\bin\\bash.exe",
    "rootPath": "C:\\Users\\Usuario\\AppData\\Local\\sf\\client\\2.1.7-355d833",
    "pluginVersions": [
      "@oclif/plugin-autocomplete 2.3.2 (core)",
      "@oclif/plugin-commands 2.2.18 (core)",
      "@oclif/plugin-help 5.2.13 (core)",
      "@oclif/plugin-not-found 2.3.31 (core)",
      "@oclif/plugin-plugins 3.1.6 (core)",
      "@oclif/plugin-search 0.0.20 (core)",
      "@oclif/plugin-update 3.1.26 (core)",
      "@oclif/plugin-version 1.3.7 (core)",
      "@oclif/plugin-warn-if-update-available 2.0.43 (core)",
      "@oclif/plugin-which 2.2.25 (core)",
      "@salesforce/cli 2.1.7 (core)",
      "apex 2.3.6 (core)",
      "auth 2.8.9 (core)",
      "data 2.5.1 (core)",
      "deploy-retrieve 1.16.0 (core)",
      "info 2.6.31 (core)",
      "limits 2.3.26 (core)",
      "login 1.2.20 (core)",
      "org 2.9.25 (core)",
      "schema 2.3.20 (core)",
      "settings 1.4.20 (core)",
      "sobject 0.1.37 (core)",
      "source 2.10.25 (core)",
      "telemetry 2.2.3 (core)",
      "templates 55.5.5 (core)",
      "trust 2.4.32 (core)",
      "user 2.3.24 (core)",
      "sfdmu 4.29.5 (user)",
      "sfdx-git-delta 5.24.1 (user)"
    ]
  },
  "sfdxEnvVars": [],
  "sfEnvVars": [
    [
      "SF_AUTOUPDATE_DISABLE",
      "true"
    ],
    [
      "SF_BINPATH",
      "C:\\Users\\Usuario\\AppData\\Local\\sf\\client\\bin\\sf"
    ],
    [
      "SF_DISABLE_AUTOUPDATE",
      "true"
    ],
    [
      "SF_REDIRECTED",
      "1"
    ],
    [
      "SF_UPDATE_INSTRUCTIONS",
      "Use \"npm update --global @salesforce/cli\" to update npm-based installations."
    ]
  ],
  "cliConfig": {
    "_base": "@oclif/core@2.9.4",
    "debug": 0,
    "topicSeparator": " ",
    "warned": false,
    "commandPermutations": {
      "validPermutations": {}
    },
    "topicPermutations": {
      "validPermutations": {}
    },
    "_commands": {},
    "_topics": {},
    "root": "C:\\Users\\Usuario\\AppData\\Local\\sf\\client\\2.1.7-355d833",
    "name": "@salesforce/cli",
    "version": "2.1.7",
    "channel": "stable",
    "valid": true,
    "arch": "x64",
    "platform": "win32",
    "windows": true,
    "bin": "sf",
    "binAliases": [
      "sfdx"
    ],
    "nsisCustomization": "scripts/nsis.nsi",
    "dirname": "sf",
    "flexibleTaxonomy": true,
    "userAgent": "@salesforce/cli/2.1.7 win32-x64 node-v18.15.0",
    "shell": "C:\\Program Files\\Git\\usr\\bin\\bash.exe",
    "home": "C:\\Users\\Usuario",
    "cacheDir": "C:\\Users\\Usuario\\AppData\\Local\\sf",
    "configDir": "C:\\Users\\Usuario\\AppData\\Local\\sf",
    "dataDir": "C:\\Users\\Usuario\\AppData\\Local\\sf",
    "errlog": "C:\\Users\\Usuario\\AppData\\Local\\sf\\error.log",
    "binPath": "C:\\Users\\Usuario\\AppData\\Local\\sf\\client\\bin\\sf",
    "nodeEngine": ">=14.0.0"
  },
  "pluginSpecificData": {
    "@salesforce/plugin-source": [
      {
        "sourceApiVersion": "56.0",
        "targetDevHubApiVersion": "58.0",
        "targetOrgApiVersion": "58.0"
      }
    ]
  },
  "diagnosticResults": [
    {
      "testName": "salesforcedx plugin not installed",
      "status": "pass"
    },
    {
      "testName": "no linked plugins",
      "status": "pass"
    },
    {
      "testName": "[@salesforce/plugin-source] sourceApiVersion matches apiVersion",
      "status": "pass"
    },
    {
      "testName": "[@salesforce/plugin-source] default target DevHub max apiVersion matches default target org max apiVersion",
      "status": "pass"
    },
    {
      "testName": "[@salesforce/plugin-source] sourceApiVersion matches default target org max apiVersion",
      "status": "warn"
    },
    {
      "testName": "using latest or latest-rc CLI version",
      "status": "pass"
    }
  ],
  "suggestions": [
    "Check https://github.com/forcedotcom/cli/issues for CLI issues posted by the community.",
    "Check http://status.salesforce.com for general Salesforce availability and performance.",
    "The sourceApiVersion in sfdx-project.json doesn't match the max apiVersion of the default target org. As a result, you're not using the latest features available in API version 58.0. The version mismatch isn't a problem, as long as it's the behavior you actually want."
  ],
  "logFilePaths": [],
  "commandExitCode": 0
}
```

**Comment by shetzel** [comment]
You have an old version of sf (`@salesforce/cli/0.0.43`) still installed on your machine.  I would completely uninstall everything following [these instructions](https://developer.salesforce.com/docs/atlas.en-us.sfdx_setup.meta/sfdx_setup/sfdx_setup_move_to_sf_v2.htm) and if necessary, [these instructions](https://developer.salesforce.com/docs/atlas.en-us.sfdx_setup.meta/sfdx_setup/sfdx_setup_uninstall.htm).

Basically, when everything is uninstalled properly there will be nothing reported by: `which sfdx` and `which sf`.

Please reopen if this does not solve your issue.
